### PR TITLE
Add CentOS 8 Stream-based OKD 4.10 ironic image builds

### PR DESCRIPTION
**Pull request by LorbusChris**
This commit adds CentOS 8 Stream-based image builds of all ironic
components for Baremetal IPI support in OKD 4.10

Note that for `ironic-static-ip-manager` and `ironic-machine-os-downloader` images the definition from the `Dockerfile.ocp` file in the respective repos can be used, making Prow replace the FROM directive from the file with a CentOS 8 Stream image. All other images have (or will have, see blockers below) their own dedicated `Dockerfile.okd` definitions.

Blocked by: https://github.com/openshift/ironic-agent-image/pull/21 https://github.com/openshift/ironic-hardware-inventory-recorder-image/pull/510
Fixes: https://github.com/openshift/okd/issues/197

/cc @vrutkovs

### CODE REVIEW

**Comment by LorbusChris** [comment]
Seems I gotta investigate why the image build says the base is not a CentOS stream image. 

/hold

**Comment by zaneb** [comment]
ironic-image Dockerfile has merged.
/retest-required

File: core-services/release-controller/_releases/release-ocp-4.10-ci.json [hunk 1]
```diff
@@ -25,6 +25,8 @@
           "ovirt-installer",
           "sig-storage-local-static-provisioner",
           "ironic",
+          "ironic-agent",
+          "ironic-hardware-inventory-recorder",
```
Comment (LorbusChris) [inline core-services/release-controller/_releases/release-ocp-4.10-ci.json:29]: Context: we don't want to mirror `ironic-hardware-inventory-recorder` out to the `origin` namespace from `ocp`. It's not being used, and if it were we'd want to promote an OKD-specific version to `origin` anyway.

**Review by elfosardo — CHANGES_REQUESTED** [review]
with this config in place the jobs will run every time there is a change, even if it's not related to okd or doesn't change the actual build
we should limit the run as it's done for the other jobs for ocp

**Comment by LorbusChris** [comment]
@elfosardo this is only image builds, not any unit or e2e tests. I think we want to ensure the image gets rebuilt for each change. I don't think image build jobs for OCP are limited in any way either, otherwise I'd appreciate a pointer :)

**Comment by elfosardo** [comment]
@LorbusChris we'd like to avoid having the OKD image rebuilt if there are changes *only* to OCP related files (dockerfile or package list) or to text files (e.g. README, LICENSE)
please have a look at the OCP jobs for an example, you can find more info at https://docs.ci.openshift.org/docs/how-tos/contributing-openshift-release/#tolerated-changes-to-generated-jobs

**Comment by LorbusChris** [comment]
@elfosardo ahh yea, I've added appropriate `skip_if_only_changed` to the generated OKD jobs, and also adapted the OCP ones accordingly.

**Comment by vrutkovs** [comment]
/approve
/lgtm

assuming this looks good to metal folks

**Comment by LorbusChris** [comment]
/hold cancel
Approval from @smarterclayton is still required.

File: core-services/image-mirroring/openshift/mapping_origin_4_10 [hunk 2]
```diff
@@ -130,8 +130,11 @@ registry.ci.openshift.org/origin/4.10:ibm-cloud-controller-manager quay.io/opens
...
 registry.ci.openshift.org/origin/4.10:ironic-agent quay.io/openshift/origin-ironic-agent:4.10 quay.io/openshift/origin-ironic-agent:4.10.0 quay.io/openshift/origin-ironic-agent:latest
-registry.ci.openshift.org/origin/4.10:ironic-hardware-inventory-recorder quay.io/openshift/origin-ironic-hardware-inventory-recorder:4.10 quay.io/openshift/origin-ironic-hardware-inventory-recorder:4.10.0 quay.io/openshift/origin-ironic-hardware-inventory-recorder:latest
+registry.ci.openshift.org/origin/4.10:ironic-ipa-downloader quay.io/openshift/origin-ironic-ipa-downloader:4.10 quay.io/openshift/origin-ironic-ipa-downloader:4.10.0 quay.io/openshift/origin-ironic-ipa-downloader:latest
+registry.ci.openshift.org/origin/4.10:ironic-machine-os-downloader quay.io/openshift/origin-ironic-machine-os-downloader:4.10 quay.io/openshift/origin-ironic-machine-os-downloader:4.10.0 quay.io/openshift/origin-ironic-machine-os-downloader:latest
+registry.ci.openshift.org/origin/4.10:ironic-static-ip-manager quay.io/openshift/origin-ironic-static-ip-manager:4.10 quay.io/openshift/origin-ironic-static-ip-manager:4.10.0 quay.io/openshift/origin-ironic-static-ip-manager:latest
```
[the remark this answers is not in the document]
Comment (smarterclayton) [reply core-services/image-mirroring/openshift/mapping_origin_4_10:137]: ok

**Pull request merged**